### PR TITLE
Update target sdk

### PR DESCRIPTION
**Pull request by onyxvd**
Update target SDK and dependencies to newer versions

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -61,38 +58,31 @@ android {
...
-    compile 'com.squareup.okhttp3:okhttp-urlconnection:3.4.1'
+    implementation 'com.squareup.retrofit2:retrofit:2.4.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.4.0'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.10.0'
+    implementation 'com.squareup.okhttp3:okhttp-urlconnection:3.4.1'
 
     // https://github.com/SimonVT/schematic
-    apt 'net.simonvt.schematic:schematic-compiler:0.6.3'
-    compile 'net.simonvt.schematic:schematic:0.6.7'
+    annotationProcessor 'net.simonvt.schematic:schematic-compiler:0.6.3'
```
Comment (groupsky) [inline app/build.gradle]:
```suggestion
    // TODO: migrate to v0.7.x
    annotationProcessor 'net.simonvt.schematic:schematic-compiler:0.6.3'
```

File: app/src/main/java/org/bspb/smartbirds/pro/SmartBirdsApplication.java [hunk 2]
```diff
@@ -29,6 +29,9 @@
 public class SmartBirdsApplication extends Application {
 
     public static final String TAG = "SBP";
+
+    public static final String AUTH_FILES = BuildConfig.APPLICATION_ID + ".SmartBirdsFileProvider";
```
Comment (groupsky) [inline app/src/main/java/org/bspb/smartbirds/pro/SmartBirdsApplication.java]:
```suggestion
    public static final String FILES_AUTHORITY = BuildConfig.APPLICATION_ID + ".SmartBirdsFileProvider";
```

File: app/src/main/java/org/bspb/smartbirds/pro/ui/fragment/NewEntryPicturesFragment.java [hunk 3]
```diff
@@ -176,6 +179,10 @@ public void onEventMainThread(@SuppressWarnings("UnusedParameters") ImageFileCre
     public void onEventMainThread(ImageFileCreated event) {
         currentImage = new ImageStruct(event.imageFileName, event.imagePath, event.uri);
         Intent intent = new Intent(INTENT_TAKE_PICTURE).putExtra(MediaStore.EXTRA_OUTPUT, event.uri);
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
+            intent.setClipData(ClipData.newRawUri("", event.uri));
+            intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION|Intent.FLAG_GRANT_READ_URI_PERMISSION);
```
Comment (groupsky) [inline app/src/main/java/org/bspb/smartbirds/pro/ui/fragment/NewEntryPicturesFragment.java]:
```suggestion
            intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
```

File: app/build.gradle [hunk 4]
```diff
@@ -61,38 +58,31 @@ android {
...
 dependencies {
-    apt "org.androidannotations:androidannotations:4.0.0"
-    compile "org.androidannotations:androidannotations-api:4.0.0"
+    annotationProcessor "org.androidannotations:androidannotations:4.4.0"
+    implementation "org.androidannotations:androidannotations-api:4.4.0"
 
-    compile fileTree(dir: 'libs', include: ['*.jar'])
-    compile 'com.google.android.gms:play-services-maps:9.6.1'
-    compile 'de.greenrobot:eventbus:2.4.1'
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
[the remark this answers is not in the document]
Comment (onyxvd) [reply app/build.gradle]: we are still using some jar files

**Pull request merged**